massgov/mayflower
# React/dp 10134  refactor backstop scenarios

## Description
Updates the mf/react Backstop implementation to discover all stories by parsing each story module to find the `kind` and `storyName` values instead of assuming they will match the path and module dir name.

Additional changes/improvements:
* added better documentation in /mayflower/react/docs/visual-regression.md
* added backstop:approve npm run command to facilitate update workflow (doc updated to use npm run commands mostly instead of docker-compose directly)
* incorporated Steven's suggestion to use path.resolve for backstop paths
* added support for a --debug flag to help with local debugging

## Related Issue / Ticket
- [JIRA issue https://jira.mass.gov/browse/DP-10134](https://jira.mass.gov/browse/DP-10134)

## Steps to Test
Tests in CI should pass for this branch -- other pending PRs will need to update their reference images after this is merged.

If you would like to test locally:
1. cd react
1. git checkout react/DP-10134--refactor-backstop-scenarios
1. npm install
1. npm run backstop # all tests should pass, run takes ~3min
1. edit a component to test for a failure. eg. remove the className prop in Button
1. npm run backstop # the Button atom test should fail
1. open backstop/data/html_report/index.html

## Screenshots
N/A

## Additional Notes:
N/A

#### Impacted Areas in Application


* react backstop tests

#### @TODO


*

#### Today I learned...
